spec/controllers/applications_controller_spec.rb

```diff
@@ -0,0 +1,214 @@
+    end
+
+    context 'if admin' do
+      before { sign_in_admin }
+      it { is_expected.to be_success }
+      it { is_expected.to have_http_status(200) }
+    end
+
+    context 'if no user' do
+      it { is_expected.to have_http_status(302) }
+    end
+  end
+
+  describe 'post #CREATE' do
+    let(:app_attrs) { { application: attributes_for(:application) } }
+    subject { post :create, app_attrs }
+
+    context 'if user' do
+      before { sign_in_user }
+      it { expect { subject }.to change { Application.count }.by 1 }
+    end
+
+    context 'if admin' do
+      before { sign_in_admin }
+      it { expect { subject }.to change { Application.count }.by 1 }
+    end
+
+    context 'if no user' do
+      it { expect { subject }.to change { Application.count }.by 0 }
+    end
+  end
+
+  describe 'get #NEW' do
+    subject { get :new }
+
+    context 'if user' do
+      before { sign_in_user }
+      it { expect(subject).to be_success }
+      it { is_expected.to have_http_status(200) }
+    end
+
+    context 'if admin' do
+      before { sign_in_admin }
+      it { expect(subject).to be_success }
+      it { is_expected.to have_http_status(200) }
+    end
+
+    context 'if no user' do
+      it { expect(subject).not_to be_success }
+      it { is_expected.to have_http_status(302) }
+    end
+  end
+
+  describe 'get #SHOW' do
+    def show_app app
+      get :show, id: app.id
+    end
+
+    context 'if user' do
+      before { sign_in_user }
+
+      it 'should show user app' do
+        expect(show_app user_application).to have_http_status(200)
+      end
+
+      xit 'should not show admin app' do
+        expect(show_app admin_application).to have_http_status(302)
+      end
+    end
+
+    context 'if admin' do
+      before { sign_in_admin }
+
+      it 'should show user app' do
+        expect(show_app user_application).to have_http_status(200)
+      end
+
+      it 'should show admin app' do
+        expect(show_app admin_application).to have_http_status(200)
+      end
+    end
+
+    context 'if no user' do
+      it 'should not show user app' do
+        expect(show_app user_application).to have_http_status(302)
+      end
+
+      it 'should not show admin app' do
+        expect(show_app admin_application).to have_http_status(302)
+      end
+    end
+  end
+
+  describe 'post #DESTROY' do
+    def destroy_app app
+      post :destroy, id: app.id
+    end
+
+    context 'if user' do
+      before { sign_in_user }
+
+      it 'should destroy user application' do
+        expect { destroy_app user_application }.to change { Application.count }.by -1
+      end
+
+      xit 'should not destroy admin application' do
+        expect { destroy_app admin_application }.to change { Application.count }.by 0
+      end
+    end
+
+    context 'if admin' do
+      before { sign_in_admin }
+
+      it 'should destroy user application' do
+        expect { destroy_app user_application }.to change { Application.count }.by -1
```

Comment:
Ambiguous negative number operator. Parenthesize the method arguments if it's surely a negative number operator, or add a whitespace to the right of the `-` if it should be a subtraction.